lms/djangoapps/support/views/program_enrollments.py

```diff
@@ -71,48 +76,36 @@ def post(self, request):
             }
         )
 
-    @staticmethod
-    def _validate_and_link(program_uuid_string, linkage_text):
+
+class LinkProgramEnrollmentSupportAPIView(GenericAPIView):
+    """
+    Support-only API View for changing learner enrollments by support
+    staff
+    """
+    authentication_classes = (
+        JwtAuthentication, SessionAuthentication
+    )
+    permission_classes = (IsAuthenticated,)
+
+    @method_decorator(require_support_permission)
+    def post(self, request):
         """
-        Validate arguments, and if valid, call `link_program_enrollments`.
+        Creates and changes learner enrollments by support staff
 
-        Returns: (successes, errors)
-            where successes and errors are both list[str]
+        * Example Request:
+            - POST /support/link_program_enrollments_details/
         """
-        if not (program_uuid_string and linkage_text):
-            error = (
-                "You must provide both a program uuid "
-                "and a series of lines with the format "
-                "'external_user_key,lms_username'."
-            )
-            return [], [error]
-        try:
-            program_uuid = UUID(program_uuid_string)
-        except ValueError:
-            return [], [
-                f"Supplied program UUID '{program_uuid_string}' is not a valid UUID."
-            ]
-        reader = csv.DictReader(
-            linkage_text.splitlines(), fieldnames=('external_key', 'username')
-        )
-        ext_key_to_username = {
-            (item.get('external_key') or '').strip(): (item['username'] or '').strip()
-            for item in reader
+
+        program_uuid = request.POST.get('program_uuid', '').strip()
+        text = request.POST.get('text', '')
```

Comment:
nit: we can use a meaningful variable name here instead of text.

Reply:
Updated to `username_pair_text`